### ISSUE TITLE
The valid_elements ! configuration is not working as expected

### ISSUE THREAD

**Issue by rafdulaf** [tinymce/tinymce#5793]
In the documentation of the "valid_elements" configuration, here https://www.tiny.cloud/docs/configure/content-filtering/#valid_elements
We can find the following :

_! | Enables removal of elements with no attributes such as <span>. They can still have content though._

So I configure a tinymce example with the following configuration: 
**valid_elements: "#p,span[annotation]"**

**The expected behavior is the following:**
The source code `<p>Test <span>of</span> test</p>` should become `<p>Test of test</p>`
The source code `<p>Test <span annotation="1">of</span> test</p>` should remain untouched

**The current behavior is the following:**
The source code `<p>Test <span>of</span> test</p>` should become `<p>Test of test</p>` **OK**
The source code `<p>Test <span annotation="1">of</span> test</p>`become `<p>Test of test</p>` **KO**

**How to reproduce**
1. Open the following editor https://codepen.io/cohcon/pen/KKVNEPM?editable=true%3Dhttps%3A%2F%2Fwww.tiny.cloud%2Fdocs%2Fdemo%2Fbasic-example%2F
2. Use the "<>" source button to view the source code
3. Add `<span annotation="1">test</span>` in a paragraph
4. Save the source
5. Reopen the source
6. The text "test" remains without its surrounding "span" **KO**

**Versions**
Tested on tinyMce 5.3.1 using Chrome 83

**Note**
I am using tinyMCE for a long time, and the bug is there since the "!" flag was introduced in v4.
So I use modified versions of tinyMCE to fix this.
Here are my modifications in the function `function Schema(settings) {` 
* change `var elementRuleRegExp = /^([#+\-])?([^\[!\/]+)(?:\/([^\[!]+))?(?:(!?)\[([^\]]+)\])?$/,` to `var elementRuleRegExp = /^([#+\-!])?([^\[!\/]+)(?:\/([^\[!]+))?(?:(!?)\[([^\]]+)\])?$/,` by adding '!' in the first group
* change `if (matches[4] === '!') {` to `if (prefix === '!') {` so it complies with the other cases in the same function
Then it works as expected

**Comment by lostkeys** [comment]
Sorry, I can't replicate your issue. With your codepen using chrome 83, the span with annotation attribute is preserved after saving the dialog and opening it again. I've tested with both Chrome and Safari on mac, and on different versions of TinyMCE. They all work as you expect.

**Comment by rafdulaf** [comment]
Indeed, I cannot not reproduce it either with the simplified example I gave you. My bad
I work on the example agin and come back to you soon

**Comment by rafdulaf** [comment]
Ok, I'm back.
The example is now ok to reproduce the issue.

**Comment by lostkeys** [comment]
Alright, i'm able to reproduce it. It sounds like a bug.

**Comment by MarkoNV** [comment]
@rafdulaf workaraound for your case would be (if I understood correctly):
`valid_elements: "#p,span[!annotation]"`

That would be even more appropriate because this requires that annotation attribute must exists (it can be empty, but must exists), while your configuration requires **any** attribute.

**Comment by rafdulaf** [comment]
@MarkoNV No you did'nt get it :) 
I do not want to force the attribute to exists. I want to remove the parent <span> is it has no attribute.

The docs says it is possible, but there is a bug in the code. I give the fix for the code. It works for years now.
